tests/test_oauth.py

```diff
@@ -124,8 +124,8 @@ def test_exchange_token(self):
         self.assertRaises(AuthException, oauth.exchange_token, "")
         self.assertRaises(AuthException, oauth.exchange_token, None)
 
-        with patch("requests.get") as mock_get:
-            mock_get.return_value.ok = False
+        with patch("requests.post") as mock_post:
```

Comment:
If you change all requests to posts where are the mock of the get? (which was the actual problem we tried to solve)

Reply:
The issue (in some of the places here was that the post wasnt mocked (as it mock the get) and therefor it try to perform the real post call and got to prod..

Reply:
But the problem were GET calls that goes to our production server. Which means that get calls were not mocked as well.

Reply:
Yea the Get was for fetching keys and now its not even get there (and also we did some posts request that just didnt generate errors but it happened..) .
also now if someone will (for example) init a Auth object without provide a public key (where now its not exist in our tests) it will do a Get request to 127.0.0.1..

Reply:
Ok, so need to make sure we have tests that also do get and the entire flow, just with mocks.

Reply:
Ok I added one magiclink test (we dont have to duplicate it for all the other as well) that cover the full flow (including Gt request for fetching the public key)).